modules/account-lib/src/coin/algo/assetTransferBuilder.ts

```diff
@@ -1,26 +1,188 @@
 import { BaseCoin as CoinConfig } from '@bitgo/statics';
-import { NotImplementedError } from '../baseCoin/errors';
-import { BaseKey } from '../baseCoin/iface';
+import BigNumber from 'bignumber.js';
+import algosdk from 'algosdk';
+import { BaseAddress } from '../baseCoin/iface';
+import { InvalidTransactionError } from '../baseCoin/errors';
+import { TransactionType } from '../baseCoin';
 import { TransactionBuilder } from './transactionBuilder';
 import { Transaction } from './transaction';
+import { AssetTransferTxnSchema } from './txnSchema';
 
 export class AssetTransferBuilder extends TransactionBuilder {
+  private _assetIndex: number;
+  private _assetAmount: number | bigint;
+  private _receiver: string;
+  private _closeTo?: string;
```

Comment:
Subclassing won't work due to the nature of how `buildImplementation` works. 

https://github.com/BitGo/BitGoJS/blob/BG-31382-feat/algo-tokens/modules/account-lib/src/coin/algo/transferBuilder.ts#L34

Build implementaiton needs to be called last. If we subclass asset transfer builder to transfer builder, then transfer builder will overwrite the txn.

Reply:
The reason I suggest to derive from TransferBuilder is because asset transfer is specialization of regular transfer, with addition assetId and a different transaction type.
Can we have a method `internalBuildImplementation()` which is called from `TransferBuilder.buildImplementation()`? This method shall be overridden in `assetTransfer`.

Alternatively, if the approach cannot be implemented for some reason - let us have the property and method names in sync between AssetTransferBuilder and TransferBuilder.